src/stores/api.ts

```diff
@@ -106,6 +106,25 @@ export const defineAPIStore = defineStore('api', {
             });
             return buildings;
         },
+        vehiclesByDispatchCenter: (state): Record<number, Vehicle[]> => {
+            const dispatchCenters: Record<number, Vehicle[]> = {};
+            state.buildings.forEach(building => {
+                const dispatchId = building.leitstelle_building_id;
+
+                // do not process dispatch centers
+                if (!dispatchId) return;
+
+                if (!dispatchCenters.hasOwnProperty(dispatchId))
+                    dispatchCenters[dispatchId] = [];
+
+                state.vehicles
+                    .filter(v => v.building_id === building.id)
+                    .forEach(vehicle => {
+                        dispatchCenters[dispatchId].push(vehicle);
+                    });
```

Comment:
I'm very concerned about performance here. How about iterating over vehicles (instead of buildings) as the outer loop and remember the dispatch-centers of buildings:
smth like that:
```pseudo
building_leitstelle_zuordnungs = {}
vehicles_by_leitstelle = {}
for each vehicle:
  if vehicle.building.leitstelle is not known then
    ind out vehicle.building.leitstelle and store in building_leitstelle_zuordnungs
  add vehicle to the vehicles_by_leitstelle
```